### PR TITLE
Revert "Revert Rust 1.68.0 upgrade"

### PR DESCRIPTION
**Pull request by ryoqun**
Reverts solana-labs/solana#30897

I want to use https://doc.rust-lang.org/nightly/std/sync/struct.Arc.html#method.into_inner, which is merged upstream after 1 day after our current nightly rustc...

cc: @yihau @ilya-bobyr

### CODE REVIEW

**Comment by ilya-bobyr** [comment]
Seems like the CI is showing the failure that caused the https://github.com/solana-labs/solana/pull/30897 revert:
https://buildkite.com/solana-labs/solana/builds/93497#01875a84-c8a7-4fa3-adb1-ecf0793c956d/1701
I could not reproduce it locally, for some reason.

File: geyser-plugin-manager/src/geyser_plugin_manager.rs [hunk 1]
```diff
@@ -386,6 +386,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
```
Comment (ilya-bobyr) [inline geyser-plugin-manager/src/geyser_plugin_manager.rs:389]: Is it OK that geyser plugins could be broken on master?

Comment (ryoqun) [reply geyser-plugin-manager/src/geyser_plugin_manager.rs:389]: nope.. ;)

I'll try to repro this as well later.

**Comment by ryoqun** [comment]
phew, finally found some time to work on this...

good news: newer nightly fixed (no extra work for us, yay)


> I could not reproduce it locally, for some reason.

odd. i could easily repro this  on my machine:

```
$ ./scripts/coverage.sh test_geyser_reload
...
running 1 test
error: test failed, to rerun pass `-p solana-geyser-plugin-manager --lib`

Caused by:
  process didn't exit successfully: `/home/sol/work/solana-labs/nightly-geyser-coverage/target/cov/debug/deps/solana_geyser_plugin_manager-39729159b0399d3b test_geyser_reload --nocapture` (signal: 11, SIGSEGV: invalid memory reference)
Failed: 101
^^^ +++
--- grcov
+ grcov target/cov/tmp --llvm --ignore '*.cargo*' --ignore '*build.rs' --ignore 'bench-tps*' --ignore 'upload-perf*' --ignore 'bench-streamer*' --ignore 'local-cluster*' -t html -o target/cov/lcov-local
+ grcov target/cov/tmp --llvm --ignore '*.cargo*' --ignore '*build.rs' --ignore 'bench-tps*' --ignore 'upload-perf*' --ignore 'bench-streamer*' --ignore 'local-cluster*' -t lcov -o target/cov/lcov.info
+ cd target/cov
+ tar zcf report.tar.gz lcov-local
-rw-r--r-- 1 sol users 101795 Apr 19 13:17 target/cov/lcov-local/index.html

```

@yihau could you bump nightly to `2023-04-19` by creating yet another docker image, which now passes on the same machine which can repro the segv:

```patch
$ git diff
diff --git a/ci/rust-version.sh b/ci/rust-version.sh
index 501d2a2bd0..2b43b606de 100644
--- a/ci/rust-version.sh
+++ b/ci/rust-version.sh
@@ -29,7 +29,7 @@ fi
 if [[ -n $RUST_NIGHTLY_VERSION ]]; then
   nightly_version="$RUST_NIGHTLY_VERSION"
 else
-  nightly_version=2023-01-22
+  nightly_version=2023-04-19
 fi

```

**Comment by ilya-bobyr** [comment]
Dependabot now [needs 1.68 or a newer](https://github.com/dependabot/dependabot-core/pull/7066) toolchain.
[A PR](https://github.com/dependabot/dependabot-core/pull/7122) that is supposed to add a nice error is still pending.
For now, dependabot is failing with multiple error that are not very descriptive, if a repo is pinned to any toolchain before 1.68.
So, until we upgrade to 1.68 our dependency updated process is also stalled.

**Comment by ilya-bobyr** [comment]
Is there a reason we want to upgrade both stable and nightly version of Rust in the same PR?
Would it be acceptable to merge the stable update and work on the failures in nightly?

If it is acceptable, consider this PR: https://github.com/solana-labs/solana/pull/31276

**Comment by ilya-bobyr** [comment]
> > Is there a reason we want to upgrade both stable and nightly version of Rust in the same PR? Would it be acceptable to merge the stable update and work on the failures in nightly?
> > If it is acceptable, consider this PR: #31276
> 
> Usually we do both since we can, but it's not required. I've been bumping stable some of the last few times without bumping nightly due to memory layout changes that needed to be resolved.

From the standpoint of bisection and reverting changes, it is better to split things that do not have to be connected.

Not sure if there is some time saving optimization in the image generation process.
As it is manual, I guess, it is easier to do it once, rather than twice.

---

I think it is possible to create GitHub actions that [create Docker images](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action) and [publish it](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images), when files change.
Not sure if this automation is strictly necessary here.
But it would remove the image creation manual step, making it possible to do the full update process via GitHub for anyone with repo write permissions.

**Comment by yihau** [comment]
we can make it automatically. The reason why I haven't implemented it cuz I have some ideas for our ci environment. If the improvement happen, we don't need it anymore. I can adjust the priority if you feel impeded!

**Comment by ilya-bobyr** [comment]
> we can make it automatically. The reason why I haven't implemented it cuz I have some ideas for our ci environment. If the improvement happen, we don't need it anymore. I can adjust the priority if you feel impeded!

Oh, no, I do not have any immediate need for it :)
I was thinking out loud I guess %)

It seems that the CI scripts will actually install the right toolchain, even if it is not available.  Though having it in the image probably saves a few minutes in the CI run.

**Comment by ryoqun** [comment]
closing; there should be a pr with latest nightly rustc soon as all the blocking issues are now resolved. cc: @yihau

**Pull request closed without merging**